### PR TITLE
Change prefixes again

### PR DESCRIPTION
**Pull request by oxarbitrage**
For issue https://github.com/ghost-coin/ghost-private/issues/26

As we discussed in chat we will have the prefix `ghost` for the mainnet, `tghost` for the testnet and `rghost` for the mainnet.
I think we are interested only on the addresses generated(please correct me if i am wrong) so we can keep the other prefixes the same or simple. This PR changes the testnet and the regtest address prefixes. It does not change the mainnet(due to additional test failure that should be researched in a new PR).

The same tests as https://github.com/ghost-coin/ghost-private/pull/25 were done, all passing except the ones that were failing before prefix changes. 

Let me know what do you think.

### CODE REVIEW

File: src/test/data/key_io_invalid.json [hunk 1]
```diff
@@ -6,147 +6,147 @@
...
-        "HawnNccY4eAftYiXHivq4G5PrgrDBjV8jA8FkdmP5Mz339g3HEAZA686wPMVCFv6B3L45V4dQsv"
+        "7uY1911cjBhUBU47FrmsjHfgR5TpyKfLmEw46oQgtmB7zaVsj7nM"
     ],
     [
-        "tc1qffxp8qa79nzajxlu6alrj7nv4ks645ly4ejk4m"
+        "2GU5w1KFnmdqYAmE9zAYEGLc1ti8wj4AenoUdgfGWKyyw3vcnJwj3VMGgtY8UujXT7MCng1UfAGvZ"
     ],
     [
-        "tgstw137swvxnn988c55rr4afcf498x79w0ty99l5s3au0f425w9mtmxa7s2h499c"
+        "TC1QUWAKVFSHPPRJJ35E0M420CMYCP2SEDFYZ4F38S"
```
Comment (TheQuantumPhysicist) [inline src/test/data/key_io_invalid.json:21]: Why are some addresses shorter, and some longer? Seems chaotic. Can we explain this behavior?

Comment (akshaynexus) [reply src/test/data/key_io_invalid.json:21]: This is random on the json file,not a real issue as it is intended

Comment (TheQuantumPhysicist) [reply src/test/data/key_io_invalid.json:21]: @akshaynexus Please check and approve this PR so that I could merge.

File: src/chainparams.cpp [hunk 2]
```diff
@@ -828,7 +828,7 @@ class CRegTestParams : public CChainParams {
         base58Prefixes[EXT_PUBLIC_KEY_BTC] = {0x04, 0x35, 0x87, 0xCF}; // tpub
         base58Prefixes[EXT_SECRET_KEY_BTC] = {0x04, 0x35, 0x83, 0x94}; // tprv
 
-        bech32Prefixes[PUBKEY_ADDRESS].assign       ("tph",(const char*)"tph"+3);
+        bech32Prefixes[PUBKEY_ADDRESS].assign       ("rghost",(const char*)"rghost"+6);
```
Comment (TheQuantumPhysicist) [inline src/chainparams.cpp:831]: When the time comes to change this or the mainnet's, use the same map you see for testnet. This is not safe in C++ and can cause buffer overflow.

**Review by akshaynexus — APPROVED** [review]

**Pull request merged**